### PR TITLE
TAPI exception to problems mapping exposure

### PR DESCRIPTION
**Pull request by reinsch82**
Fixes https://github.com/gradle/gradle-private/issues/4095

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test ACT

File: platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/exec/BuildOutcomeReportingBuildActionRunner.java [hunk 1]
```diff
@@ -66,6 +71,7 @@ public Result run(BuildAction action, BuildTreeLifecycleController buildControll
 
         buildLogger.logResult(result.getBuildFailure());
         new TaskExecutionStatisticsReporter(styledTextOutputFactory).buildFinished(taskStatisticsCollector.getStatistics());
+        problemsService.reportMapping(result.getBuildFailure());
```
Comment (reinsch82) [inline platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/exec/BuildOutcomeReportingBuildActionRunner.java:74]: only report problems that are related to the exception causing the failure

File: platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java [hunk 2]
```diff
@@ -73,4 +77,24 @@ public InternalProblemReporter getInternalReporter() {
...
+        ImmutableMap.Builder<Throwable, Collection<Problem>> builder = ImmutableMap.<Throwable, Collection<Problem>>builder();
+
+        Throwable cause = throwable;
+
+        // TODO: handle MultiCauseException
+        // TODO: handle cyclic causes (ie. this == cause and this.cause.cause == this)
+
+        while (cause != null) {
+            for (Map.Entry<Throwable, Collection<Problem>> entry : problemsForThrowables.asMap().entrySet()) {
+                if (entry.getKey().equals(cause)) {
```
Comment (wolfs) [inline platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java:92]: 🤔 Why not call `problemsForThrowables.get(cause)`?

Comment (reinsch82) [reply platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblems.java:92]: good question :)
I'll fix that.

**Comment by reinsch82** [comment]
@bot-gradle test this

**Comment by reinsch82** [comment]
@bot-gradle test ACTR

**Pull request closed without merging**